### PR TITLE
Sms javaedits

### PR DESCRIPTION
**Pull request by spacedsweden**

### CODE REVIEW

File: docs/sms/sms-java.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+![requestbin request](images\requestbin-request.png)
+
+### Handle incoming SMS with Java pring boot
+
+Below is a spring controller. 
+
+```java
+
+@RestController
```
Comment (thongtran715) [inline docs/sms/sms-java.md:95]: Why this one in Java, but below ngrok is using js ?

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md [hunk 2]
```diff
@@ -38,27 +36,24 @@ ApiConnection connection = ApiConnection.builder()
...
-```java
-import com.clxcommunications.xms.ClxApi;
-
-connection.createBatch(ClxApi.batchTextSms()
-        .sender("ignored")
-        .addRecipient("{RECIPIENT_1_PHONE_NUMBER}")
-        .addRecipient("{RECIPIENT_2_PHONE_NUMBER}")
-        .body("Hello from Sinch!")
-        .build());
+import com.sinch.xms.api;
```
Comment (thongtran715) [inline docs/tutorials/tutorials-java/tutorial-java-send-sms.md:39]: Should be `import com.sinch.xms.SinchSMSApi;`

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md [hunk 3]
```diff
@@ -38,27 +36,24 @@ ApiConnection connection = ApiConnection.builder()
...
+String SERVICE_PLAN_ID = "{YOUR_SERVICE_PLAN_ID}";
```
Comment (thongtran715) [inline docs/tutorials/tutorials-java/tutorial-java-send-sms.md:40]: The complete code is 
`package example;

import com.sinch.xms.ApiConnection;
import com.sinch.xms.SinchSMSApi;
import com.sinch.xms.api.GroupResult;
import com.sinch.xms.api.MtBatchTextSmsResult;

public class Example {

  private static final String SERVICE_PLAN_ID = "SERVICE_PLAN_ID";
  private static final String TOKEN = "SERVICE_TOKEN";
  private static final String[] RECIPIENTS = {"1232323131", "3213123"};
  private static final String SENDER = "SENDER";

  public static void main(String[] args) {
    try (ApiConnection conn =
        ApiConnection.builder().servicePlanId(SERVICE_PLAN_ID).token(TOKEN).start()) {

      // Sending a simple Text Message
      MtBatchTextSmsResult batch =
          conn.createBatch(
              SinchSMSApi.batchTextSms()
                  .sender(SENDER)
                  .addRecipient(RECIPIENTS)
                  .body("Something good")
                  .build());

      System.out.println("Successfully sent batch " + batch.id());

      // Creating simple Group
      GroupResult group = conn.createGroup(SinchSMSApi.groupCreate().name("Subscriber").build());

      // Adding members (numbers) into the group
      conn.updateGroup(
          group.id(), SinchSMSApi.groupUpdate().addMemberInsertion("15418888", "323232").build());

      // Sending a message to the group
      batch = conn.createBatch(
          SinchSMSApi.batchTextSms()
              .addRecipient(group.id().toString())
              .body("Something good")
              .build());

      System.out.println("Successfully sent batch " + batch.id());
    } catch (Exception e) {
      System.out.println("Batch send failed: " + e.getMessage());
    }
  }
}
`

Comment (thongtran715) [inline docs/tutorials/tutorials-java/tutorial-java-send-sms.md:40]: The code is 
```
package example;

import com.sinch.xms.ApiConnection;
import com.sinch.xms.SinchSMSApi;
import com.sinch.xms.api.GroupResult;
import com.sinch.xms.api.MtBatchTextSmsResult;

public class Example {

  private static final String SERVICE_PLAN_ID = "SERVICE_PLAN_ID";
  private static final String TOKEN = "SERVICE_TOKEN";
  private static final String[] RECIPIENTS = {"1232323131", "3213123"};
  private static final String SENDER = "SENDER";

  public static void main(String[] args) {
    try (ApiConnection conn =
        ApiConnection.builder().servicePlanId(SERVICE_PLAN_ID).token(TOKEN).start()) {

      // Sending a simple Text Message
      MtBatchTextSmsResult batch =
          conn.createBatch(
              SinchSMSApi.batchTextSms()
                  .sender(SENDER)
                  .addRecipient(RECIPIENTS)
                  .body("Something good")
                  .build());

      System.out.println("Successfully sent batch " + batch.id());

      // Creating simple Group
      GroupResult group = conn.createGroup(SinchSMSApi.groupCreate().name("Subscriber").build());

      // Adding members (numbers) into the group
      conn.updateGroup(
          group.id(), SinchSMSApi.groupUpdate().addMemberInsertion("15418888", "323232").build());

      // Sending a message to the group
      batch = conn.createBatch(
          SinchSMSApi.batchTextSms()
              .addRecipient(group.id().toString())
              .body("Something good")
              .build());

      System.out.println("Successfully sent batch " + batch.id());
    } catch (Exception e) {
      System.out.println("Batch send failed: " + e.getMessage());
    }
  }
}


```

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md [hunk 4]
```diff
@@ -106,21 +101,21 @@ try (connection) {
...
 
-To wrap up, here's the complete sources of a minimal Java application that starts a connection to the Sinch REST API, sends a message then closes the connection.
+To wrap up, here's the complete sources of a minimal Java application that starts a connection to the Sinch REST API, sends a message, then closes the connection.
 
 ```java
-import com.clxcommunications.xms.ApiConnection;
-import com.clxcommunications.xms.ApiException;
-import com.clxcommunications.xms.ClxApi;
-import com.clxcommunications.xms.api.MtBatchTextSmsResult;
+import com.sinch.sms.ApiConnection;
```
Comment (thongtran715) [inline docs/tutorials/tutorials-java/tutorial-java-send-sms.md:115]: Should be `import com.sinch.xms.ApiConnection;`

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md [hunk 5]
```diff
@@ -106,21 +101,21 @@ try (connection) {
...
+import com.sinch.sms.ApiException;
+import com.sinch.sms.api;
+import com.sinch.sms.api.MtBatchTextSmsResult;
```
Comment (thongtran715) [inline docs/tutorials/tutorials-java/tutorial-java-send-sms.md:118]: `import com.sinch.xms.api.MtBatchTextSmsResult;`

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md [hunk 6]
```diff
@@ -106,21 +101,21 @@ try (connection) {
...
+To wrap up, here's the complete sources of a minimal Java application that starts a connection to the Sinch REST API, sends a message, then closes the connection.
 
 ```java
-import com.clxcommunications.xms.ApiConnection;
-import com.clxcommunications.xms.ApiException;
-import com.clxcommunications.xms.ClxApi;
-import com.clxcommunications.xms.api.MtBatchTextSmsResult;
+import com.sinch.sms.ApiConnection;
+import com.sinch.sms.ApiException;
+import com.sinch.sms.api;
```
Comment (thongtran715) [inline docs/tutorials/tutorials-java/tutorial-java-send-sms.md:117]: remove this

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md [hunk 7]
```diff
@@ -106,21 +101,21 @@ try (connection) {
...
-To wrap up, here's the complete sources of a minimal Java application that starts a connection to the Sinch REST API, sends a message then closes the connection.
+To wrap up, here's the complete sources of a minimal Java application that starts a connection to the Sinch REST API, sends a message, then closes the connection.
 
 ```java
-import com.clxcommunications.xms.ApiConnection;
-import com.clxcommunications.xms.ApiException;
-import com.clxcommunications.xms.ClxApi;
-import com.clxcommunications.xms.api.MtBatchTextSmsResult;
+import com.sinch.sms.ApiConnection;
+import com.sinch.sms.ApiException;
```
Comment (thongtran715) [inline docs/tutorials/tutorials-java/tutorial-java-send-sms.md:116]: `import com.sinch.xms.ApiException;`

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 2 above)
Comment (JEllis72) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:39]: Change made.

File: docs/sms/sms-java.md (hunk 1 above)
Comment (JEllis72) [reply docs/sms/sms-java.md:95]: @spacedsweden - can you help with this one?

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 3 above)
Comment (JEllis72) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:40]: @thongtran715 - should this code go in the wrap up section of the document?

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 4 above)
Comment (JEllis72) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:115]: changed line 115 from import com.sinch.sms. ApiConnection; to import com.sinch.xms.ApiConnection;

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 3 above)
Comment (thongtran715) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:40]: Yes, this is a complete code sending sms messages in Java way

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 5 above)
Comment (JEllis72) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:118]: changed line 118 from import com.sinch.sms.api.MtBatchTextSmsResult; to import com.sinch.xms.api.MtBatchTextSmsResult;

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 6 above)
Comment (JEllis72) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:117]: @thongtran715 - can you confirm exactly what I am to remove? Is it just line 123 shown in red here (which I cannot find in the file), or is it lines 115, 116 and 117?

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 7 above)
Comment (JEllis72) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:116]: Am I replacing import com.sinch.sms.ApiException with import com.sinch.xms.ApiException; ?

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 6 above)
Comment (thongtran715) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:117]: sorry, removed 117

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 7 above)
Comment (thongtran715) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:116]: ah , yes

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 6 above)
Comment (JEllis72) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:117]: I have removed line 117

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 7 above)
Comment (JEllis72) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:116]: I have replaced import com.sinch.sms.ApiException with import com.sinch.xms.ApiException;

File: docs/tutorials/tutorials-java/tutorial-java-send-sms.md (hunk 3 above)
Comment (JEllis72) [reply docs/tutorials/tutorials-java/tutorial-java-send-sms.md:40]: I have added this code to the Wrap Up section

File: docs/sms/sms-java.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+            .token({your token})
+            .start()
+String [] RECIPIENTS = {"{To Number}"};
+MtBatchTextSmsResult batch =
+          conn.createBatch(
+              SinchSMSApi.batchTextSms()
+                  .sender("{your free test number}")
+                  .addRecipient(RECIPIENTS)
+                  .body("This is a test message!")
+                  .build()
```
Comment (thongtran715) [inline docs/sms/sms-java.md:40]: missing the closing parethense after `build()`

File: docs/sms/sms-java.md (hunk 1 above)
Comment (thongtran715) [reply docs/sms/sms-java.md:95]: @spacedsweden

File: docs/tutorials/tutorials-java.md [hunk 9]
```diff
@@ -4,49 +4,143 @@ excerpt: Tutorials for our Java SDK
...
+```javascript
+String SENDER = "SENDER"; // Optional
+String [] RECIPIENTS = {"1232323131", "3213123"}  ;
+MtBatchTextSmsResult batch =
+          conn.createBatch(
+              SinchSMSApi.batchTextSms()
+                  .sender(SENDER)
+                  .addRecipient(RECIPIENTS)
+                  .body("Something good")
+                  .build()
```
Comment (thongtran715) [inline docs/tutorials/tutorials-java.md:125]: Missing closing parethenese after `build()`

**Pull request merged**